doc/how_to/authentication/configuration.md

```diff
@@ -12,6 +12,9 @@ The first step in configuring a OAuth is to specify a specific OAuth provider. P
 * `gitlab`: GitLab
 * `google`: Google
 * `okta`: Okta
+* `generic`: Generic OAuth Provider with configurable endpoints
+* `password`: Generic password grant based OAuth Provider with configurable endpoints
+* `code`: Generic code challenge grant based OAuth Provider with configurable endpoints
```

Comment:
`code` might be somewhat ambiguous name, since there is also a device code flow. What about `auth_code`? Right, @armaaar ?

Reply:
I agree. There is also an OAuth flow called Device Code flow: https://oauth.net/2/grant-types/device-code/
This name might cause confusion

Reply:
Makes sense to me, thanks!

Reply:
Maybe link from the docs here or elsewhere to the API docs to make it easy to explore the details where reading this?

With search it's possible to find e.g. [panel.auth.GenericLoginHandler](https://panel.holoviz.org/api/panel.auth.html#panel.auth.GenericLoginHandler) and other relevant classes, but at least for me having a link to click when reading here would be nice to have.